support/k8s/spire-adm-webhook/webhook.go

```diff
@@ -0,0 +1,225 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"path"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spire/pkg/common/idutil"
+	"github.com/spiffe/spire/proto/spire/api/registration"
+	"github.com/spiffe/spire/proto/spire/common"
+	"github.com/zeebo/errs"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	admv1beta1 "k8s.io/api/admission/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type WebhookConfig struct {
+	Log         logrus.FieldLogger
+	R           registration.RegistrationClient
+	TrustDomain string
+	Cluster     string
+	PodLabel    string
+}
+
+type Webhook struct {
+	c WebhookConfig
+}
+
+func NewWebhook(config WebhookConfig) *Webhook {
+	return &Webhook{
+		c: config,
+	}
+}
+
+func (w *Webhook) Initialize(ctx context.Context) error {
+	// ensure there is a node registration entry for PSAT nodes in the cluster.
+	return w.createEntry(ctx, &common.RegistrationEntry{
+		ParentId: idutil.ServerID(w.c.TrustDomain),
+		SpiffeId: w.nodeID(),
+		Selectors: []*common.Selector{
+			{Type: "k8s_psat", Value: fmt.Sprintf("cluster:%s", w.c.Cluster)},
+		},
+	})
```

Comment:
If this entry already exists, Initialize will fail?

Reply:
:)